pandas/core/computation/expressions.py

```diff
@@ -104,11 +104,16 @@ def _evaluate_numexpr(op, op_str, a, b):
         a_value = a
         b_value = b
 
-        result = ne.evaluate(
-            f"a_value {op_str} b_value",
-            local_dict={"a_value": a_value, "b_value": b_value},
-            casting="safe",
-        )
+        try:
+            result = ne.evaluate(
+                f"a_value {op_str} b_value",
+                local_dict={"a_value": a_value, "b_value": b_value},
+                casting="safe",
+            )
+        except TypeError:
```

Comment:
Catching the error here, instead of higher op in `_na_arithmetic_op`, ensures that we try again with the plain standard `op` instead of directly try again with `_masked_arith_op`